sopel/tools/time.py

```diff
@@ -112,16 +150,20 @@ def get_channel_timezone(db, channel):
         return None
 
 
-def get_timezone(db=None, config=None, zone=None, nick=None, channel=None):
+def get_timezone(
```

Comment:
I see two issues with that approach:

* these functions repeat the same checks/transformations of `tz`
* it would create a hard coupling between `get_timezone` and these, which I don't really want

So for now, I'll keep the existing code as it is. If it has to change, that would be for another PR.